Spatial.gd

```diff
@@ -420,80 +425,110 @@ func gen_map_markers():
-	var selected_category_name = find_pedigree_name(category_item, category_item.get_metadata(0).get_name())
-	local_data.visible_dict[selected_category_name] = category_item.is_checked(1)
-	switch_checkboxes(category_item, category_item.is_checked(1))
-	
-	node_checkmark_switch(selected_category_name, category_item, self.paths)
-	node_checkmark_switch(selected_category_name, category_item, self.minimap)
-	node_checkmark_switch(selected_category_name, category_item, self.icons)
-
-	$Control/LocalData.save_local()
+	Settings.local_category_data[category_item.get_metadata(0)] = {
+		"checked" : category_item.is_checked(1),
+	} 
+	update_node_visibility(category_item, self.paths)
+	update_node_visibility(category_item, self.icons)
 
+	Settings.save()
 
-func find_pedigree_name(category_item: TreeItem, pedigree_name: String):
-	var parent =  category_item.get_parent()
-	if parent.is_selectable(0):
-		pedigree_name = parent.get_metadata(0).get_name() + "." + pedigree_name
-		pedigree_name = find_pedigree_name(parent, pedigree_name)
-	return pedigree_name
-
-
-func node_checkmark_switch(selected_category_name: String, category_item, nodes):
+func update_node_visibility(category_item: TreeItem, nodes):
 	for node in nodes.get_children():
-		if node.category_name.begins_with(selected_category_name):
-			local_data.visible_dict[node.category_name] = category_item.is_checked(1)
-			if category_item.is_checked(1):
-				node.show()
+		var node_name = node.waypoint.get_category().get_name()
+		if node_name.begins_with(category_item.get_metadata(0)):
+			if is_category_visible(category_item):
+				node.visible = true
 			else:
-				node.hide()
+				node.visible = false
+			if node.get_name() == "Path":
+				var index = node.get_index()
+				var route2d = self.minimap.get_child(index)
+				route2d.visible= node.visible 
+
+#Child visibility is contigent on all parents having permission 
+func is_category_visible(category_item: TreeItem) -> bool:
+	if category_item == marker_packs.get_root():
+		return true
+	if category_item.is_checked(1):
+		return is_category_visible(category_item.get_parent())
+	else:
+		return false
 
 
-func switch_checkboxes(item: TreeItem, checked: bool):
-	if item.get_cell_mode(1) == TreeItem.CELL_MODE_CHECK:
-		item.set_checked(1, checked)
-		local_data.visible_dict[find_pedigree_name(item, item.get_metadata(0).get_name())] = checked
-		
-	var child_item = item.get_children()
+func show_or_hide_all(is_visible: bool) :
+	if is_visible == false:
+		Settings.local_category_data.clear()
+	var category_item = self.marker_packs.get_root().get_children()
+	while category_item != null:
+		category_item.set_checked(1, is_visible)
+		change_all_categories(category_item, is_visible)
+		update_node_visibility(category_item, self.paths)
+		update_node_visibility(category_item, self.icons)
+		category_item = category_item.get_next()
+	Settings.save()
+
+
+func change_all_categories(category_item: TreeItem, is_visible: bool):
```

Comment:
This function is only  used for iterating over `show_or_hide_all`'s children. It is also pretty tightly tied to to `show_or_hide_all` so maybe change this function name to `_show_or_hide_all` to show that it is directly linked. Also it is strange that we are doing the first level of iteration in the `show_or_hide_all` function, iterating over the marker packs, but then all the other levels of iteration  within this recursive function.

Reply:
As discussed. Show and Hide All are being removed from the program for now.